web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -46,8 +88,68 @@ describe("<SidebarCardFundingNudge />", () => {
     card = generateSidebarCardContent({ id: fundingNudge });
   });
 
+  const generateTaxFilingUserData = (params: {
+    publicFiling: boolean;
+    formedInNavigator?: boolean;
+    businessName?: string;
+    responsibleOwnerName?: string;
+    taxId?: string;
+  }): UserData => {
+    let formationData: FormationData = generateEmptyFormationData();
+
+    const legalStructureId = params.publicFiling
+      ? randomPublicFilingLegalType()
+      : randomLegalStructure({ requiresPublicFiling: false }).id;
+    if (params.publicFiling) {
+      formationData = generateFormationData(
+        {
+          completedFilingPayment: !!params.formedInNavigator,
+          getFilingResponse: generateGetFilingResponse({ success: params.formedInNavigator }),
+        },
+        legalStructureId as FormationLegalType
+      );
+    }
+
+    return generateUserDataForBusiness(
+      generateBusiness({
+        profileData: generateProfileData({
+          legalStructureId: legalStructureId,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayTaxAccessButton;
+            })
+          ).id,
+          taxId: params.taxId ? `*${params.taxId.slice(1)}` : "",
+          encryptedTaxId: params.taxId ? `encrypted-${params.taxId}` : "",
+          businessName: params.businessName || "",
+          responsibleOwnerName: params.responsibleOwnerName || "",
+        }),
+        formationData: formationData,
+        taxFilingData: generateTaxFilingData({ state: undefined }),
+      })
+    );
+  };
+
   describe("when clicking funding button for non-generic industry", () => {
```

Comment:
[boulder] We should talk about the testing strategy here. We're mocking the API response in one of the tests, but we're not actually doing anything to assert that logic is executing aside from making the tests pass.